### ISSUE TITLE
Using the Algolia Search

### ISSUE THREAD

**Issue by chriscoyier** [cdnjs/cdnjs#12683]
Hey team!

Not exactly an "issue" here so feel free to close and send me elsewhere. I wasn't sure who to reach out to about general use of the project itself.

At CodePen, we've been trying to make it easy for people to find and use CDNjs URL's for years. We've gone through different takes on it, from downloading the entire JSON output and using it on our side, to using the API. It's been a bit tricky, as part it is trying to make it *fast* and *return great results*, and other issue is scoping the results to either **CSS** or **JavaScript**.

One way forward might be using Algolia. We've spoken with them about it and of course they are happy to be of service. Y'all already have Algolia search running on your homepage. 

Basically, my ask is: can we get in on that? Can we use your Algolia index as well to power an in-CodePen CDNjs search? 

I can be reached at chris@codepen.io or here or [on Twitter](https://twitter.com/chriscoyier) or whatever.

**Comment by PeterDaveHello** [comment]
@chriscoyier thanks for reaching, let me think about how to deal with it, feel free to DM me on Twitter ;)

**Comment by chriscoyier** [comment]
Hey hey!

For a basic **step one** here, I don't think we need anything from you at all, other than permission to use your Algolia index. 

The **scoping by CSS or JS** feature I think we can perhaps handle client-side or get Algolia's help to do in there somehow.

**Comment by PeterDaveHello** [comment]
If you need Algolia index here, it's fine to use it ;)

**Comment by chriscoyier** [comment]
Cool thanks! I'll close this.

I'd say for future development it'd still be kinda neat to see the API or somehow search in some fashion offer that distinction between CSS and JS results. But I'm just one vote ;)

**Comment by PeterDaveHello** [comment]
@chriscoyier pull requests are welcome ;)

**Comment by PeterDaveHello** [comment]
@chriscoyier just wanna confirm with you that if it's acceptable that the searching behavior will be changed on CodePen? 

As there are 400M+ files, the size of metadata is huge, so we haven't indexed all the files on Algolia but only saves the "main file" of each libraries' latest version, but our API provides the whole file list among all versions, so currently, users can search by assets' filename among all the libraries.

So the difference here would be: Algolia is faster, but CDNJS API provides more complete info.

**Comment by chriscoyier** [comment]
Yep everything is good on our end. I didn't even notice that the Algolgia search had more limited info. What we need I guess is more limited. We just need fast high quality results, and the "main file" is all we need. Type "reac" and the main UMD React file should come up, and it totally does.

This isn't live yet, but seems to be working fantastically (video):

https://d.pr/v/WsU9Na

**Comment by PeterDaveHello** [comment]
@chriscoyier cool! Will this change directly replace the old one, or there would be a beta/testing period? Thanks!

**Comment by chriscoyier** [comment]
It will replace the old one. It's not ready for beta just yet as it's going out with other related features, but we _might_ have a beta before it goes out for everyone, not sure!

**Comment by PeterDaveHello** [comment]
Okay. Please let me know when there is beta ;)
BTW, is it also fine that our Algolia index only contains the "latest" version of each library?

**Comment by chriscoyier** [comment]
Yep that's ideal for us.

**Comment by PeterDaveHello** [comment]
@chriscoyier okay, cool, thanks!
